evalml/pipelines/components/transformers/samplers/base_sampler.py

```diff
@@ -170,20 +171,33 @@ def _initialize_oversampler(self, X, y, sampler_class):
             y (pd.Series): Target features
             sampler_class (imblearn.BaseSampler): The sampler we want to initialize
         """
-        _, y_ww = self._prepare_data(X, y)
+        _, y_pd = self._prepare_data(X, y)
         sampler_params = {
             k: v
-            for k, v in copy.copy(self.parameters).items()
-            if k not in ["sampling_ratio", "sampling_ratio_dict"]
+            for k, v in self.parameters.items()
+            if k not in ["sampling_ratio", "sampling_ratio_dict", "k_neighbors_default"]
         }
         if self.parameters["sampling_ratio_dict"] is not None:
             # make the dictionary
-            dic = self._convert_dictionary(self.parameters["sampling_ratio_dict"], y_ww)
+            dic = self._convert_dictionary(self.parameters["sampling_ratio_dict"], y_pd)
         else:
             # create the sampling dictionary
             sampling_ratio = self.parameters["sampling_ratio"]
-            dic = make_balancing_dictionary(y_ww, sampling_ratio)
+            dic = make_balancing_dictionary(y_pd, sampling_ratio)
         sampler_params["sampling_strategy"] = dic
+
+        # check for k_neighbors value
+        neighbors = self.parameters["k_neighbors_default"]
+        min_counts = y_pd.value_counts().values[-1]
+        if min_counts <= 1:
```

Comment:
Nit pick, but this could only be... one, right? Otherwise it wouldn't show up in `value_counts` to begin with?

Reply:
Hahah yeah, this is true!